### PR TITLE
Add support to detect the word frequency

### PR DESCRIPTION
**Pull request by p2635**
For Issue #66. the PR is super messy because I am actually a newbie programmer who has never really dealt with Anki addon development at all nor Python. By super messy, I got the code to somehow work after about 3 hours of trial and error. I had to keep copying and pasting the addon folder to my local Anki install because I don't know how to properly set up a dev env for developing anki addons.

The work is basically based on:
https://github.com/ernop/anki-chinese-word-frequency

Upon entering a word, it will fetch the frequency and fill in a 'Frequency field' to one of below:
-very basic (1st 500 words)
-basic (500-1000)
-very common (1000-2000)
-common (2000-3700)
-uncommon (3700-6300)
-rare (6300-10500)
-very rare (10500-18600)
-obscure (18600-50000)

I'm sure things can be moved around to consts and many things can be done to improve. Sorry for the mess, I would love to learn though. Feedback is welcome. I'm surprised I got so far messing around on my own though, I really needed this feature!

### CODE REVIEW

**Comment by luoliyan** [comment]
This is a really good effort, but we'll try to clean it up a bit before merging. I'll add some comments.

File: chinese/freq.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    slevels=sorted(levels.items(),key=lambda x:x[0]*-1)
+    words=sorted([w.upper() for w in levels.values()],key=lambda x:-1*len(x))
+
+    # Declare path to the corpus file
+    corpus_path = join(dirname(realpath(__file__)), 'lib', 'num' , 'internet-zh.num')
+
+    # Navigate to corpus file and open it or throw error if it doesn't exist
+    if os.path.exists(corpus_path):
+        blob=codecs.open(corpus_path,'r','utf8').read()
```
Comment (luoliyan) [inline chinese/freq.py]: No need for `codecs` in Python 3. Just use `open`.

File: chinese/freq.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        25:'common',
+        13:'uncommon',
+        7:'rare',
+        2:'very rare',
+        0:'obscure'}
+
+    slevels=sorted(levels.items(),key=lambda x:x[0]*-1)
+    words=sorted([w.upper() for w in levels.values()],key=lambda x:-1*len(x))
+
+    # Declare path to the corpus file
```
Comment (luoliyan) [inline chinese/freq.py]: I don't really believe in comments, which I realise is heresy. If the code is clear enough, don't comment. If it's not clear, rewrite it. Here, it's clear enough.

File: chinese/freq.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    corpus_path = join(dirname(realpath(__file__)), 'lib', 'num' , 'internet-zh.num')
+
+    # Navigate to corpus file and open it or throw error if it doesn't exist
+    if os.path.exists(corpus_path):
```
Comment (luoliyan) [inline chinese/freq.py]: I would drop this check entirely. We'll be bundling the file with the addon, right? If so, just assume it's always there.

File: chinese/freq.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        blob=codecs.open(corpus_path,'r','utf8').read()
+    else:
+        showInfo('you need to copy the internet-zh.num file to %s , or edit the source file to point to the file.'%corpus_path)
+
+    # Main logic here
+    pat=re.compile('(.+ '+hanzi+')\n')
+    try:
+        res=pat.findall(blob)[0]
+    except:
```
Comment (luoliyan) [inline chinese/freq.py]: Just check if `res is None`.

File: chinese/freq.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        return 'Not found'
+    description=''
+    frequency_html=''
+    if res and type(res)!=tuple:
```
Comment (luoliyan) [inline chinese/freq.py]: Maybe I'm a bit rusty. Is `res` ever not `None` and also not a tuple? My guess is once we've checked for `None`, this check becomes redundant.

**Comment by p2635** [comment]
@luoliyan thank you so much, this will help me improve on my skills. I want to become contributor when I'm good enough. I will review the comments when I'm back home.

File: chinese/freq.py (hunk 1 above)
Comment (luoliyan) [reply chinese/freq.py]: Also no need for 'r', since that's the default.

Comment (p2635) [reply chinese/freq.py]: Solved on  df8181e

**Comment by p2635** [comment]
Hi @luoliyan , regarding the last 2 comments I'm afraid that's beyond my knowledge since I didn't write the original code

UPDATE: never mind, it looks like you've cleaned it up!

**Pull request merged**